iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert_slice.mlir

```diff
@@ -16,7 +16,7 @@ func @insert_slice_convert
 //   CHECK-DAG:   %[[C4:.+]] = constant 4
 //   CHECK-DAG:   %[[DIM0:.+]] = tensor.dim %[[ARG0]], %[[C0]]
 //       CHECK:   %[[UPDATE:.+]] = flow.tensor.update %[[ARG1]], %[[ARG0]][%[[C4]], %[[C2]], %[[C0]]]
-//  CHECK-SAME:     : tensor<1x4x48xf32> -> tensor<?x24x48xf32>{%[[DIM0]]}
+//  CHECK-SAME:     : tensor<1x4x48xf32> -> %[[ARG0]] as tensor<?x24x48xf32>{%[[DIM0]]}
```

Comment:
If I understand correctly, the shape after `as` is not the shape of the SSA value before the `as`, i.e. its a place to say that the shape of the result is different than the shape of the tied operand?
I see the value in it, but its a bit of a lift to internalize it.

Reply:
This was needed here because we need both the argument name again (for tying) and the type - previously the tying was implicit and that was confusing to read in the IR as you would have to know the op is tied - this just spells out that the result is tied. Since the first $target use is untyped there's no way for the types to differ (as the type printed is type($target)).